woocommerce/woocommerce
# [Product Block Editor] Add post_password parameter to the Woo product REST api

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Add post_password for products for REST API V3

Closes #39128 .



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Open the legacy product editor at Product > Add New.
2. Create any type of product and publish it, or pick an existing one (without password)
3. Open the console and type:
```js
wp.data.dispatch( 'wc/admin/products' ).updateProduct( <post_id>, { post_password: 'test_password' });
```
4. Open the product in the legacy product editor and you should see that the password was added:
<img width="317" alt="image" src="https://github.com/woocommerce/woocommerce/assets/13437655/58aeb677-fa11-4d6b-9cbd-0f17284a4bca">

5. Open the console and type:
```js
wp.data.select( 'wc/admin/products' ).getProduct(<post_id>)
```
6. If the result is not yet cached, it will return undefined. You can either check the response from the network tab, or run the exact same command again. Either way, you should see the correct `post_password` field in the result.




### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
